schema/crdb/3.0.1/up.sql

```diff
@@ -0,0 +1,42 @@
+BEGIN;
+
+SELECT CAST(
+    IF(
+        (
+            SELECT version = '3.0.0' and target_version = '3.0.1'
+            FROM omicron.public.db_metadata WHERE singleton = true
+        ),
+        'true',
+        'Invalid starting version for schema change'
+    ) AS BOOL
+);
+
+ALTER TABLE omicron.public.ip_pool
+    DROP COLUMN IF EXISTS project_id,
+    ADD COLUMN IF NOT EXISTS is_default BOOLEAN NOT NULL DEFAULT FALSE,
+    DROP CONSTRAINT IF EXISTS project_implies_silo,
+    DROP CONSTRAINT IF EXISTS internal_pools_have_null_silo_and_project;
+
+COMMIT;
+
+-- needs to be in its own transaction because of this thrilling bug
+-- https://github.com/cockroachdb/cockroach/issues/83593
```

Comment:
would a nested transaction also work? that would get rid of the duplicate schema version checks